### ISSUE TITLE
Bypass Sentinel file check with a "none" policy

### ISSUE THREAD

**Issue by BarthV** [kubernetes-sigs/cluster-api-provider-kubevirt#233]
## Context

Following last Syncup Meeting (Tuesday April 18), we are now targeting multiple new features :

1. short term : bypassing Sentinel file check with a "none" policy
2. mid term : working on a alternate qemu-guest-agent usage to check file
3. long term : making the whole SSH logic optional (for keygen & cloud-init management...) with something like a Sentinel File check mode selector in spec

This Issue tracks short term **item 1.**

## Feature Description

Define & implement a Sentinel Check file mode selector.
By default, this selector must preserve current CAPK behavior (SSH check)
We need to add a new "None" mode, which bypass Sentinel file check and act like machine were always bootstrapped.

* SSH keys are still managed and provisionned all the time on the VMs, even if Sentinel file is not checked.
* Configuration should be tested & flexible enough for future enhancements (Actions items 2. & 3. )

## CRD Change Proposal

Adding a new optional string field in KubevirtMachine (& KubevirtMachineTemplate) CR.
This string would describe how [CAPI Sentinel File](https://cluster-api.sigs.k8s.io/developer/providers/bootstrap.html#sentinel-file) is checked for a single Machine.
Since this value is set on the machine, multiple check strategies can live in the same cluster without any problem.
Sanity checks and default value should be implemented to filter values & to make SSH the default option.

Expected value would be : `none` and `ssh`. Empty value will fallback to `ssh`, unexpected values would fails the `KubevirtMachine` reconciliation (or maybe fallback to `ssh` too , but it really can be risky and deceiving for the end user).

**New `KubevirtMachine.spec` field :**
```yaml
virtualMachineBootstrapCheck:
  checkStrategy: [ "none" | "ssh" ]
```

**Example in a template :**
```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
kind: KubevirtMachineTemplate
metadata:
  name: capi-cluster1-cp
  namespace: capi
spec:
  template:
    spec:
      virtualMachineBootstrapCheck:
        checkStrategy: none
      virtualMachineTemplate:
        metadata:
          namespace: capi
        spec:
          runStrategy: Always
          template:
            spec:
              terminationGracePeriodSeconds: 60
              evictionStrategy: External
              [...]
```

**Comment by BarthV** [comment]
`virtualMachineBootstrapCheck` could also be named `virtualMachineBootstrap`. This is probably shorter.

**Comment by BarthV** [comment]
fixed by #239
